libafl/src/state/mod.rs

```diff
@@ -705,7 +698,32 @@ where
 
         self.continue_loading_initial_inputs_custom(fuzzer, executor, manager, forced, loader)
     }
-
+    fn load_file<E, EM, Z>(
+        &mut self,
+        path: &PathBuf,
+        manager: &mut EM,
+        fuzzer: &mut Z,
+        executor: &mut E,
+        forced: bool,
+        loader: &mut dyn FnMut(&mut Z, &mut Self, &Path) -> Result<I, Error>,
+    ) -> Result<(), Error>
+    where
+        E: UsesState<State = Self>,
+        EM: EventFirer<State = Self>,
+        Z: Evaluator<E, EM, State = Self>,
+    {
+        log::info!("Loading file {:?} ...", &path);
+        let input = loader(fuzzer, self, path)?;
+        if forced {
+            let _: CorpusId = fuzzer.add_input(self, executor, manager, input)?;
+        } else {
+            let (res, _) = fuzzer.evaluate_input(self, executor, manager, input)?;
```

Comment:
this line is the one that can exit early (which will mean the fuzzer restarts form the top, after loading the current `State`)